### PR TITLE
Add "All Wallets" to wallet dropdown options on the `Transactions` page

### LINKED ISSUE #305: the general wallets page should have "all wallets" as part of the dropdown options
Screenshot from 2023-12-04 11-21-49
Screenshot from 2023-12-04 11-22-04

### PR DESCRIPTION
**Pull request by ukane-philemon**
This PR renames the nav tab `Wallets` to `All Wallets` and updates the `Transactions` tab to display all wallets' txs by default. Closes #305 

Also, some minor refactoring and UI changes are included in this PR.
Before(the completed icon was too close to the date and did not align with the hidden asset info view):
Screenshot 2023-12-07 at 11 13 27 PM
After:
Screenshot 2023-12-07 at 11 20 48 PM

### CODE REVIEW

File: ui/page/info/info_page.go [hunk 1]
```diff
@@ -422,7 +422,7 @@ func (pg *WalletInfo) listenForNotifications() {
 	}
 
 	txAndBlockNotificationListener := &sharedW.TxAndBlockNotificationListener{
-		OnTransaction: func(transaction *sharedW.Transaction) {
+		OnTransaction: func(walletID int, transaction *sharedW.Transaction) {
```
Comment (dreacot) [inline ui/page/info/info_page.go:425]: why do we need to pass the wallet id here

Comment (ukane-philemon) [reply ui/page/info/info_page.go:425]: Saw a todo that required ignoring a tx if the wallet selected is different from the wallet that owns the tx.

**Review by JustinBeBoy — CHANGES_REQUESTED** [review]
if you update some UI on transaction page, please update margin for icon and text on dropdown
image


also I get app freeze when select the wallet on dropdown, please debug it

b2

File: ui/page/transaction/transactions_page.go [hunk 2]
```diff
@@ -491,7 +490,16 @@ func (pg *TransactionsPage) HandleUserInteractions() {
...
+		assetIndex := pg.walletDropDown.SelectedIndex()
+		// The "All Wallets" dropdown item is the first in the dropdown list.
+		if assetIndex == 0 {
+			pg.selectedWallet = nil
+		} else {
+			// Assets added as dropdown items have an index of
+			// actuallyIndex+1 due to the "All Wallets" dropdown item.
+			assetIndex--
+			pg.selectedWallet = pg.assetWallets[assetIndex]
+		}
```
[the remark this answers is not in the document]
Comment (ukane-philemon) [reply ui/page/transaction/transactions_page.go:502]: Can you throw more light on this?

File: ui/page/info/info_page.go (hunk 1 above)
Comment (ukane-philemon) [reply ui/page/info/info_page.go:425]: Please see: https://github.com/crypto-power/cryptopower/pull/313/files#r1420727718

File: ui/page/transaction/transactions_page.go [hunk 3]
```diff
@@ -523,29 +532,33 @@ func (pg *TransactionsPage) HandleUserInteractions() {
 }
 
 func (pg *TransactionsPage) listenForTxNotifications() {
-	// Listen for ntfns for all wallets. But ignore ntfns if the wallet sending the ntfn is not the currently selected wallet.
```
[the remark this answers is not in the document]
Comment (dreacot) [reply ui/page/transaction/transactions_page.go:526]: Alright got it

File: libwallet/assets/btc/txandblocknotifications.go [hunk 4]
```diff
@@ -96,9 +96,9 @@ func (asset *Asset) AddTxAndBlockNotificationListener(txAndBlockNotificationList
 	}
 
 	asset.txAndBlockNotificationListeners[uniqueIdentifier] = &sharedW.TxAndBlockNotificationListener{
-		OnTransaction: func(transaction *sharedW.Transaction) {
+		OnTransaction: func(walletID int, transaction *sharedW.Transaction) {
 			if txAndBlockNotificationListener.OnTransaction != nil {
-				go txAndBlockNotificationListener.OnTransaction(transaction)
+				go txAndBlockNotificationListener.OnTransaction(walletID, transaction)
```
Comment (dreacot) [inline libwallet/assets/btc/txandblocknotifications.go:101]:
```suggestion
				go txAndBlockNotificationListener.OnTransaction(asset.ID, transaction)
```

can't we get the id fromt he calling asset?

Comment (ukane-philemon) [reply libwallet/assets/btc/txandblocknotifications.go:101]: But I think it's better we pass it in when executing the actual function.

File: ui/page/transaction/transactions_page.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (JustinBeBoy) [reply ui/page/transaction/transactions_page.go:502]: I have researched your code, you did a great job, please ignore this comment

**Comment by JustinBeBoy** [comment]
LGTM

**Review by JustinBeBoy — APPROVED** [review]

File: libwallet/internal/politeia/politeia_client.go [hunk 5]
```diff
@@ -132,25 +132,27 @@ func (c *politeiaClient) batchProposals(tokens []string) ([]Proposal, error) {
...
+		fmt.Println("error decoding proposal metadata:", err)
+		return prop
 	}
 
 	var meta metadataProposal
 	err = json.Unmarshal(decodedBytes, &meta)
-	if err != nil {
+	if err == nil {
 		log.Error("error unmarshalling metadataProposal:", err)
+		return prop
```
Comment (dreacot) [inline libwallet/internal/politeia/politeia_client.go:146]: why are we checking if error == nil and then trying to log an error when it's nil

Comment (ukane-philemon) [reply libwallet/internal/politeia/politeia_client.go:146]: Dang! `err != nil`

**Review by dreacot — APPROVED** [review]

**Pull request merged**